Add PHP 8.0 support

bfoosness opened the pull request:
|    Q          |   A
|-------------- | ------
| New Feature   | yes

### Description

Close #6

bfoosness committed:
Update composer.json and fix unit tests.

Signed-off-by: Brent Foosness <bfoosness@users.noreply.github.com>

bfoosness committed:
Run vendor/bin/phpunit --migrate-configuration

Signed-off-by: Brent Foosness <bfoosness@users.noreply.github.com>

froschdesign commented on phpunit.xml.dist:
```diff
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <phpunit xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
-         xsi:noNamespaceSchemaLocation="vendor/phpunit/phpunit/phpunit.xsd"
+         xsi:noNamespaceSchemaLocation="https://schema.phpunit.de/9.3/phpunit.xsd"
```
Please revert this because we can use the local XML schema definition here otherwise we must update the URL with each update.

bfoosness replied:
Will do. Odd that the PHPUnit migration tool would opt to make this change instead of sticking with the local definition.

bfoosness committed:
Revert schema location.

Signed-off-by: Brent Foosness <bfoosness@users.noreply.github.com>

bfoosness commented on .travis.yml:
```diff
@@ -12,38 +12,28 @@ env:
+    - php: 8.0
       env:
         - DEPS=latest
+        - COMPOSER_ARGS="--no-interaction --ignore-platform-reqs"
```
I think we're just waiting on laminas-json, laminas-math, and laminas-servicemanager.

marc-jan committed:
added support for 'serializers' from config

Signed-off-by: Marc-Jan Barnhoorn <marc-jan@gekvansoftware.nl>

marc-jan committed:
updated AdapterPluginManagerFactoryTest

Signed-off-by: Marc-Jan Barnhoorn <marc-jan@gekvansoftware.nl>

marc-jan committed:
removed void return type

Signed-off-by: Marc-Jan Barnhoorn <marc-jan@gekvansoftware.nl>

marc-jan committed:
added plugin manager documentation

Signed-off-by: Marc-Jan Barnhoorn <marc-jan@gekvansoftware.nl>

marc-jan committed:
applied some feedback about headers in the plugin manager documentation

Signed-off-by: Marc-Jan Barnhoorn <marc-jan@gekvansoftware.nl>

marc-jan committed:
added configuration example for serializer

Signed-off-by: Marc-Jan Barnhoorn <marc-jan@gekvansoftware.nl>

marc-jan committed:
mention autoloading of plugin manager factory in MVC and Mezzio applications

Signed-off-by: Marc-Jan Barnhoorn <marc-jan@gekvansoftware.nl>

bfoosness committed:
Remove tests that are skipped with PHP >= 7.

Signed-off-by: Brent Foosness <bfoosness@users.noreply.github.com>

weierophinney committed:
qa: add GHA workflows

- Remove `.travis.yml`
- Update docs-build workflow to run on release, not push
- Add automatic releases workflow
- Add continuous integration workflow
  - Configure to add bcmath extension
  - Configure to exclude locked runs of PHP 5.6 and 7.0-7.2
- Add `composer.lock`
- Remove branch aliases from package definition
- Rename phpcs configuration file

Signed-off-by: Matthew Weier O'Phinney <matthew@weierophinney.net>

weierophinney committed:
docs: remove `CHANGELOG.md`

Changelog will now be kept in milestone descriptions, and propagated to tag and release notes.

Signed-off-by: Matthew Weier O'Phinney <matthew@weierophinney.net>

weierophinney committed:
docs: update badges in README

Signed-off-by: Matthew Weier O'Phinney <matthew@weierophinney.net>

weierophinney committed:
docs: fix linting errors

Signed-off-by: Matthew Weier O'Phinney <matthew@weierophinney.net>

weierophinney committed:
docs: minor changes in preparation for merging for 2.10.0

- one sentence per line
- reference "since 2.10.0" for new features

Signed-off-by: Matthew Weier O'Phinney <matthew@weierophinney.net>

weierophinney committed:
qa: remove obsolete exclusions from CI config

Signed-off-by: Matthew Weier O'Phinney <matthew@weierophinney.net>

weierophinney committed:
qa: updates lockfile

Signed-off-by: Matthew Weier O'Phinney <matthew@weierophinney.net>

weierophinney committed:
fix: drop support for obsolete versions of packages

- Bumps laminas-json minimum to 3.1
- Bumps laminas-stdlib version to 3.2
- Bumps laminas-math version to 3.3
- Bumps laminas-servicemanager version to 3.6

Signed-off-by: Matthew Weier O'Phinney <matthew@weierophinney.net>
